infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDiscoveryPolicy.java

```diff
@@ -0,0 +1,28 @@
+package org.corfudb.infrastructure.management;
+
+/**
+ * {@link FailureDiscoveryPolicy} is the enumeration of policies representing concrete algorithms
+ * for suspecting the failure or healing of specific servers in a Corfu cluster of servers.
+ *
+ * Created by Sam Behnam on 10/19/18.
+ */
+public enum FailureDiscoveryPolicy {
+    /**
+     * FULLY_CONNECTED_CLUSTER represents a failure and healing discovery policy in which the
+     * implementing algorithms:
+     * Suspect a node of Failed status when the given node is NOT FULLY connected to all the
+     * healthy servers in the cluster.
+     * Suspect a node of Healed status when the given node is FULLY connected to all the healthy
+     * servers in the cluster.
+     */
+    FULLY_CONNECTED_CLUSTER,
+    /**
```

Comment:
strongly consider "Centralized" vs "Decentralized" . Also, describing the algorithm in the policy seems to obscure the actually policy. Maybe describe what makes the policy centralized/decentralized

Reply:
I picked the “fully connected cluster” corresponding to “fully connected mesh” in cs and “complete graph” in math (the “centrally connected cluster” was the descriptive name for a “star topology cluster”). It is correct that “fully connected cluster” allows decentralization as there are multiple paths (no one single direct connected node) however I feel calling it decentralized cluster policy might be confusing. If you feel the other way is more clear, I’ll change that accordingly.
As for the policy, these enums are supposed of being descriptive name of the outcome of underlying algorithms. (e.g. the outcome of suspect nodes of failure/healing so that that their removal/addition lead to a complete graph or star schema correspondingly). Hence the description of outcomes of these policies. Again, if you feel otherwise, I will make it more abstract.

Reply:
Renamed the descriptive "centrally connected cluster" to centrazlied_cluster and added description on the outcome graph of the cluster after applying these policies. (i.e. complete graph vs star topology)

Reply:
I think the names should reflect the "decision making" of the algorithm, not the connectivity between the cluster nodes. Not to focus on the nomenclature too much, but the terms don't seem to fit. I also don't have any suggestions.

Reply:
I changed the document to better represent the decision making aspect of each policy each of which representing an algorithm.